### PR TITLE
CVS-58544: Model that failed to load have status END, OK instead of LOADING, ERROR

### PR DESCRIPTION
**Pull request by ficol**

### CODE REVIEW

File: src/test/modelmanager_test.cpp [hunk 1]
```diff
@@ -613,7 +613,7 @@ TEST(ModelManager, HandlingInvalidLastVersion) {
     modelInstanceUnloadGuard.reset();
     status = manager.getModelInstance(modelDirectory.name, 3, modelInstance3, modelInstanceUnloadGuard);
     modelInstanceUnloadGuard.reset();
-    ASSERT_EQ(status, ovms::StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE);
+    ASSERT_EQ(status, ovms::StatusCode::MODEL_VERSION_NOT_LOADED_YET);
```
Comment (dkalinowski) [inline src/test/modelmanager_test.cpp:616]: This status name and message indicate model is in loading phase but in fact there already was loading and it failed.

Comment (dkalinowski) [reply src/test/modelmanager_test.cpp:616]: Is this behavior expected? @atobiszei

File: src/test/modelmanager_test.cpp [hunk 2]
```diff
@@ -640,6 +641,41 @@ TEST(ModelManager, HandlingInvalidLastVersion) {
     manager.reloadModelWithVersions(config);
     ASSERT_EQ(modelInstance1->getStatus().getState(), ovms::ModelVersionState::END);
     ASSERT_EQ(modelInstance2->getStatus().getState(), ovms::ModelVersionState::AVAILABLE);
+    ASSERT_EQ(modelInstance2->getStatus().getErrorCode(), ovms::ModelVersionStatusErrorCode::OK);
+}
+
+TEST(ModelManager, HandlingInvalidVersionAtBeginning) {
```
Comment (dkalinowski) [inline src/test/modelmanager_test.cpp:647]: I'm adding unit test for that behavior here:
https://github.com/openvinotoolkit/model_server/pull/726/files#diff-d6ce9ab93bcb5115bd96af8e9a5513155abe39ce8587a97e04b7099de7b62ff5R645

We should stick to one test.

**Comment by dkalinowski** [comment]
In `ModelInstance::waitForLoaded` we wait for model to be loaded with a timeout. I suppose we should alter the logic there as well.

File: src/modelmanager.cpp [hunk 3]
```diff
@@ -990,18 +992,20 @@ Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
...
     for (const auto version : *versionsFailed) {
         SPDLOG_LOGGER_TRACE(modelmanager_logger, "Removing available version {} due to load failure.", version);
         if (std::binary_search(availableVersions.begin(), availableVersions.end(), version)) {
             availableVersions.erase(std::remove(availableVersions.begin(), availableVersions.end(), version), availableVersions.end());
         }
+        failedVersions.insert(version);
     }
     // refresh versions to retire based on failed reloads
     requestedVersions = config.getModelVersionPolicy()->filter(availableVersions);
     getVersionsToChange(config, model->getModelVersions(), requestedVersions, versionsToStart, versionsToReload, versionsToRetire);
```
Comment (atobiszei) [inline src/modelmanager.cpp:1008]: Shouldn;t this code be removed now?

File: src/test/modelmanager_test.cpp (hunk 1 above)
Comment (atobiszei) [reply src/test/modelmanager_test.cpp]: Model failed to load due to misconfiguration. Misconfiguration is expected to be fixed so it is reasonable to expect that model will finally be loaded.

**Review by rasapala — APPROVED** [review]
LGTM

**Pull request closed without merging**